cirq/google/engine/calibration.py

```diff
@@ -103,8 +109,39 @@ def __len__(self) -> int:
         return len(self._metric_dict)
 
     def __str__(self) -> str:
+        return f'Calibration(keys={list(sorted(self.keys()))})'
 
-        return 'Calibration(keys={})'.format(list(sorted(self.keys())))
+    def __repr__(self) -> str:
+        return ('cirq.google.Calibration(metric_dict='
+                f'{repr(dict(self._metric_dict))})')
+
+    @classmethod
+    def _from_json_dict_(cls, metrics, **kwargs):
+        metric_dict = {}
+        for metric in metrics:
+            metric_dict[metric['name']] = {}
+            for idx, val in enumerate(metric['targets']):
+                key = tuple(v2.grid_qubit_from_proto_id(q) for q in val)
+                metric_dict[metric['name']][key] = metric['values'][idx]
+
+        return cls(metric_dict=metric_dict)
+
+    def _json_dict_(self):
+        rtn = []
+        count = 0
+        for key in self._metric_dict:
+            metric_dict = {}
+            rtn.append(metric_dict)
```

Comment:
why not put this at the end of the loop body so it reads more idiomatically? i.e.: make empty dict, fill it in, add it to list.

Reply:
Done.